pkgs/tools/misc/heatseeker/default.nix

```diff
@@ -1,28 +1,34 @@
-{ stdenv, fetchFromGitHub, rustPlatform }:
+{ lib, fetchFromGitHub, rustPlatform, coreutils }:
 
 with rustPlatform;
 
 buildRustPackage rec {
   pname = "heatseeker";
-  version = "1.5.1";
+  version = "1.7.1";
 
   src = fetchFromGitHub {
     owner = "rschmitt";
     repo = "heatseeker";
     rev = "v${version}";
-    sha256 = "1fcrbjwnhcz71i70ppy0rcgk5crwwmbkm9nrk1kapvks33pv0az7";
+    sha256 = "1x7mdyf1m17s55f6yjdr1j510kb7a8f3zkd7lb2kzdc7nd3vgaxg";
   };
 
-  # Delete this on next update; see #79975 for details
-  legacyCargoFetcher = true;
+  buildDepends = [ coreutils ];
 
-  cargoSha256 = "0m3sxbz1iii31s30cnv1970i1mwfhl6gm19k8wv0n7zji30ayx07";
+  patchPhase = ''
+    # https://github.com/rschmitt/heatseeker/issues/42
+    # I've suggested using `/usr/bin/env stty`, but doing that isn't quite as simple
+    # as a substitution, and this works since we have the path to coreutils stty.
```

Comment:
I'd move the comment out of the patchPhase so it isn't exposed to the builder and part of the hash (e.g., put the comment on line 17).  As it stands, a change to the comment will cause a hash change and rebuild of the package.

Reply:
I don't expect that to happen much, but thanks for the tip!